src/ScrollBar/index.jsx

```diff
@@ -0,0 +1,117 @@
+import React                            from 'react';
+import PropTypes                        from 'prop-types';
+
+import { buildClassName, generateId }   from '../utils';
+import styles                           from './scrollbar.css';
+
+
+const ScrollBar = ( {
+    className,
+    cssMap,
+    id,
+    orientation,
+    onChange,
+    onMouseOut,
+    onMouseOver,
+    thumbSize,
+    scrollPos,
+    scrollMax,
+    scrollMin,
+    width
+} ) =>
+    (
+        <div
+            className    = { buildClassName( className, cssMap, { orientation } ) }
+            id           = { id }
+            onMouseEnter = { onMouseOver }
+            onMouseLeave = { onMouseOut } >
+            <input
+                className    = { cssMap.range }
+                value        = { scrollPos }
+                max          = { scrollMax }
+                min          = { scrollMin }
+                onChange     = { e => onChange( parseInt( e.target.value ) ) }
+                step         = "1"
+                style        = { { width: `${width}px`, '--thumbSize': `${thumbSize}%`, display: !thumbSize ? 'none' : 'block' } }
```

Comment:
> I think the width does not need to be explicitly set. My view is that the ScrollBar should just receive 100% width like most other components do. Besides, having a width prop is slightly confusing since it's the height that changes when the scroll bar is vertical.

 No, the width only changes depends on scroll direction "horizontal/vertical" as we manipulate the scrollBar itself using `transform: rotate( ); transform-origin` css rules.
> (btw nice use of CSS variable to style the thumb 👍 )

Thanks man 👍

Reply:
> No, the width only changes depends on scroll direction "horizontal/vertical" as we manipulate the scrollBar itself using transform: rotate( ); transform-origin css rules.

Internally yes, that is true. But the internal implementation is not relevant when defining the component’s external interface (i.e. props). 

Thinking about it from a consumer’s perspective, it’s the height that changes when the scroll bar is vertical. And it’s gonna be pretty confusing when setting the `width` prop changes the height of the component.

If this prop is needed it should be named something like `length`.

Reply:
Yeah we can go for `length` a great suggestion 👍 👍